evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -776,14 +779,27 @@ where
             }
         }
 
+        // If we are expecting a revert and we are currently reverting, we store the revert data and
+        // the address where it originated to inspect it later while matching the expected revert.
+        if self.expected_revert.is_some() && status == InstructionResult::Revert {
+            self.revert_per_address
+                .entry(b160_to_h160(call.contract))
+                .or_insert_with(HashSet::new)
+                .insert(retdata.clone().into());
+        }
+
         // Handle expected reverts
         if let Some(expected_revert) = &self.expected_revert {
             if data.journaled_state.depth() <= expected_revert.depth {
                 let expected_revert = std::mem::take(&mut self.expected_revert).unwrap();
+                let reverts_by_address = std::mem::take(&mut self.revert_per_address);
```

Comment:
Right, so if I understand correctly here, we're:
- saving the last revert seen by address
- then, taking out the hashmap from memory and passing it to `handle_expect_revert`

I'm unsure if this is the correct behavior—do we really wanna do `std::mem::take`? as this will reset the hashmap

Reply:
I want to reset the hashmap because after entering that if statement the only way to exit is after the `handle_revert` func is called and that function will either be successful because the revert matched or not which will cause a revert in the evm. In both cases, we don't need the tracked reverts by address anymore so I think it is safe to reset the data.

Reply:
Right, so do we need a hashmap in this case? because it then seems that this map can't hold more than 1 revert, or else we have a bug I believe because we're discarding reverts that we shouldn't. In this case we can probably just have something like an `Option<ExpectedRevertWithAddress`> right?